java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -231,55 +241,75 @@ class DatabaseImpl(
+    suspend fun insertOrUpdate(
+        storageKey: StorageKey,
+        data: DatabaseData.Collection,
+        dataType: DataType
+    ) = writableDatabase.useTransaction {
+        val db = this
+
+        // Fetch/create the entity's type ID.
+        val schemaTypeId = getSchemaTypeId(data.schema, db)
+
+        // Retrieve existing collection ID for this storage key, if one exists.
+        var collectionId = getCollectionId(storageKey, dataType, db)
+
+        if (collectionId == null) {
+            // Create a new collection ID and storage key ID.
+            collectionId = insert(
+                TABLE_COLLECTIONS,
+                null,
+                ContentValues().apply { put("type_id", schemaTypeId) }
+            )
+            insert(
+                TABLE_STORAGE_KEYS,
+                null,
+                ContentValues().apply {
+                    put("storage_key", storageKey.toString())
+                    put("data_type", dataType.ordinal)
+                    put("value_id", collectionId)
+                }
+            )
+        } else {
+            // Collection already exists; delete all existing entries.
+            // TODO: Don't blindly delete everything and re-insert: only insert/remove the diff.
+            delete(
+                TABLE_COLLECTION_ENTRIES,
+                "collection_id = ?",
+                arrayOf(collectionId.toString())
+            )
+        }
 
-            if (collectionId == null) {
-                // Create a new collection ID and storage key ID.
-                collectionId = insert(
-                    TABLE_COLLECTIONS,
-                    null,
-                    ContentValues().apply { put("type_id", schemaTypeId) }
-                )
-                insert(
-                    TABLE_STORAGE_KEYS,
-                    null,
-                    ContentValues().apply {
-                        put("storage_key", storageKey.toString())
-                        put("data_type", DataType.Collection.ordinal)
-                        put("value_id", collectionId)
-                    }
-                )
-            } else {
-                // Collection already exists; delete all existing entries.
-                // TODO: Don't blindly delete everything and re-insert: only insert/remove the diff.
-                delete(
-                    TABLE_COLLECTION_ENTRIES,
-                    "collection_id = ?",
-                    arrayOf(collectionId.toString())
-                )
+        // Insert all elements into the collection.
+        val content = ContentValues().apply {
+            put("collection_id", collectionId)
+        }
+        // TODO: Don't do this one-by-one.
+        data.values
+            .map { getEntityReferenceId(it, db) }
+            .forEach { referenceId ->
+                insert(TABLE_COLLECTION_ENTRIES, null, content.apply {
+                    put("value_id", referenceId)
+                })
```

Comment:
Please use "Put arguments on separate lines" (alt+enter with the text cursor on any one of the arguments)

Reply:
Done (and thanks)

Tried to figure out how to make it a warning in IntelliJ but I couldn't...